backend_plugin/src/main/evaluators/AddImportEvaluator.java

```diff
@@ -0,0 +1,104 @@
+package main.evaluators;
+
+import java.util.Iterator;
+
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.DocumentEvent;
+import org.eclipse.jface.text.source.Annotation;
+import org.eclipse.jface.text.source.IAnnotationModel;
+
+public class AddImportEvaluator extends FeatureEvaluator {
+	
+	private boolean unresolvedVariablesExist;
+	private DocumentEvent lastDocumentEvent;
+	
+	/**
+	 * Construct an AddImportEvaluator
+	 */
+	public AddImportEvaluator() {
+		this.featureID = "addImportStatementsSuggestion";
+		this.lastDocumentEvent = null;
+		this.unresolvedVariablesExist = false;
+	}
+
+	/**
+	 * Tracks the user's changes, and determines whether they are fixing an
+	 * "unresolved type" issue by manually typing the import statement
+	 * @return boolean true if the user is manually adding an import statement
+	 * 					when unresolved variables exist; false otherwise
+	 */
+	@Override
+	public boolean evaluateDocumentChanges(DocumentEvent docEvent) {
+		
+		if (this.lastDocumentEvent == null) {
+			
+			// This is the first DocumentEvent captured by the Evaluator
+			this.lastDocumentEvent = docEvent;
+		} else if (this.unresolvedVariablesExist) {
+			
+			// Check if the last two DocumentEvents represent the user typing "t"
+			// followed by " "
+			String textAdded = this.lastDocumentEvent.getText() + docEvent.getText();
+			int offsetDifference = docEvent.getOffset() - this.lastDocumentEvent.getOffset();
+			this.lastDocumentEvent = docEvent;
+			if (textAdded.equals("t ") && offsetDifference == 1) {
```

Comment:
This looks fine for now, but we need to consider what if someone copies / pastes "import". We should check if this catches that.